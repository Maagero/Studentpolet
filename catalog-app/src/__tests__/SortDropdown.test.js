import React from 'react';
import renderer from 'react-test-renderer';
import SortDropdown from '../components/sortDropdown/SortDropdown.js';
import { Provider } from 'mobx-react';
import RootStore from '../stores/RootStore';
import ApolloClient from 'apollo-client';
import { ApolloProvider } from 'react-apollo';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { createHttpLink } from 'apollo-link-http';
import fetch from 'node-fetch';

const cache = new InMemoryCache();
const client = new ApolloClient({
    link: createHttpLink({
        uri: 'http://localhost:3000/graphql',
        fetch: fetch,
    }),
    cache: cache,
});
const rootStore = new RootStore();


describe("SortDropdown component", () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(
                <ApolloProvider client={client}>
                    <Provider
                        rootStore={rootStore}
                        filterStore={rootStore.filterStore}
                        sortStore={rootStore.sortStore}
                        searchBarStore={rootStore.searchBarStore}
                        paginationStore={rootStore.paginationStore}>
                        <SortDropdown />
                    </Provider>
                </ApolloProvider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
