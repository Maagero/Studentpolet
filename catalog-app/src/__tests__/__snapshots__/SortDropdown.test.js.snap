// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortDropdown component renders correctly 1`] = `
<div
  className="dropdownContainer dropdown"
  onKeyDown={[Function]}
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="sortToggle dropdown-toggle btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Sorter etter
  </button>
</div>
`;
